### PR TITLE
copr: support FirstRow aggregate

### PR DESCRIPTION
**Pull request by BusyJay**
@shenli @coocood PTAL

### CODE REVIEW

File: src/server/coprocessor/aggregate.rs [hunk 1]
```diff
@@ -38,3 +39,23 @@ impl AggrFunc for Count {
         Ok(())
     }
 }
+
+type First = Option<Datum>;
+
+impl AggrFunc for First {
+    fn update(&mut self, mut args: Vec<Datum>) -> Result<()> {
+        if self.is_some() {
```
Comment (shenli) [inline src/server/coprocessor/aggregate.rs]: What's mean by is_some()? Will it guarantee that the column value will be saved only once?

File: tests/coprocessor/test_select.rs [hunk 2]
```diff
@@ -316,6 +316,36 @@ fn test_aggr_count() {
 }
 
 #[test]
+fn test_aggr_first() {
+    let count = 10;
+    let (mut store, mut end_point, ti) = initial_data(count);
```
Comment (shenli) [inline tests/coprocessor/test_select.rs]: Could you add some comment for the data in the table?
You can add it here or at the definition of initial_data().

File: src/server/coprocessor/aggregate.rs (hunk 1 above)
Comment (BusyJay) [inline src/server/coprocessor/aggregate.rs]: Yes. If it's been saved, it will be `Some(Datum)`, otherwise it will be `None`.

**Comment by shenli** [comment]
LGTM
@coocood PTAL

**Comment by coocood** [comment]
LGTM

**Pull request merged**